sys/xtimer/Makefile.dep

```diff
@@ -0,0 +1,31 @@
+DEFAULT_MODULE += auto_init_xtimer
+# In whatever combination of backends periph_timer will be required
+# - xtimer (only) -> periph_timer
+# - xtimer_on_ztimer -> ztimer_usec -> ztimer_periph_timer -> periph_timer
+# - ztimer_xtimer_compat -> ztimer_usec -> ztimer_periph_timer -> periph_timer
+# But requesting it here avoid an ifeq dependency that would be subject to
+# dependency inclusion order issues
+FEATURES_REQUIRED += periph_timer
+
+USEMODULE += xtimer_no_ztimer_default
+ifeq (,$(filter xtimer_no_ztimer_default,$(USEMODULE)))
```

Comment:
With this logic, is it even possible that the `ifeq` is true? Should we add `xtimer_no_ztimer_default`  as a default module instead?

Reply:
It's currently not possible, but the idea is that then I can have https://github.com/RIOT-OS/RIOT/pull/17721/commits/3c93df98533a11579bb83352354edb98cfae9252 that removes this, and a one liner can again bring things back.

Including or not the Module has too many dependencies for it to be a `DEAULT_MODULE`...

Reply:
I understand, fine with me!